File: Java/problem1464.java
class Solution {
    public int maxProduct(int[] nums) {
        int first = 0, second = 0;
        for (int n : nums) {
            if (n >= first) {
                int tmp = first;
                first = n;
                second = tmp;
            }
            else if (n >= second) {
                second = n;
            }
        }
        return (first - 1) * (second - 1);
    }
}

File: python/problem1464.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        first, second = 0, 0
        for n in nums:
            if n >= first:
                first, second = n, first
            elif n > second:
                second = n
        return (first - 1) * (second - 1)
